### PR TITLE
Add config option tenure_last_block_proposal_timeout_secs and do not allow reorgs at tenure boundary before it is exceeded

### PR DESCRIPTION
**Pull request by jferrant**
Closes https://github.com/stacks-network/stacks-core/issues/5423

### CODE REVIEW

**Review by obycode — DISMISSED** [review]
LGTM!

File: stacks-signer/src/chainstate.rs [hunk 1]
```diff
@@ -20,14 +20,15 @@ use blockstack_lib::chainstate::stacks::TenureChangePayload;
 use blockstack_lib::net::api::getsortition::SortitionInfo;
 use blockstack_lib::util_lib::db::Error as DBError;
 use clarity::types::chainstate::BurnchainHeaderHash;
+use clarity::util::get_epoch_time_secs;
```
Comment (jcnelson) [inline stacks-signer/src/chainstate.rs:23]: this should be `stacks_common::util`

Comment (jcnelson) [reply stacks-signer/src/chainstate.rs:23]: Also, the above should be `stacks_common::types::chainstate::BurnchainHeaderHash`.  Clarity merely re-exports these.

File: testnet/stacks-node/src/tests/signer/v0.rs [hunk 2]
```diff
@@ -4710,6 +4723,237 @@ fn reorg_locally_accepted_blocks_across_tenures_succeeds() {
         .expect("Timed out waiting for block acceptance of N+1'");
 }
 
+#[test]
+#[ignore]
+/// Test that signers that have accept a locally signed block N+1 built in tenure A cannot sign a block proposed during a
```
Comment (jcnelson) [inline testnet/stacks-node/src/tests/signer/v0.rs:4728]: Typo: "have accepted"

**Review by jcnelson — APPROVED** [review]
Approving for now, but I'm not sure this is guaranteed to prevent reorgs at tenure boundaries? It could be that some singers' slow networks cause them to:

* Accept and broadcast a signature on the last block of tenure N `B[N][-1]`
* Wait for the allotted timeout to pass, but never see `B[N][-1]` get broadcast and processed
* Accept and broadcast signatures for the first block of tenure N+1, `B[N+1][0]`, which conflicts with `B[N][-1]` (i.e. has the same height).

Am I understanding this right?  Is this okay?

**Comment by jferrant** [comment]
> Approving for now, but I'm not sure this is guaranteed to prevent reorgs at tenure boundaries? It could be that some singers' slow networks cause them to:
> 
>     * Accept and broadcast a signature on the last block of tenure N `B[N][-1]`
> 
>     * Wait for the allotted timeout to pass, but never see `B[N][-1]` get broadcast and processed
> 
>     * Accept and broadcast signatures for the first block of tenure N+1, `B[N+1][0]`, which conflicts with `B[N][-1]` (i.e. has the same height).
> 
> 
> Am I understanding this right? Is this okay?

It is not meant to prevent reorgs at the boundary. just to minimize them.

File: testnet/stacks-node/src/tests/signer/v0.rs [hunk 3]
```diff
@@ -4501,7 +4501,7 @@ fn locally_rejected_blocks_overriden_by_global_acceptance() {
 
 #[test]
 #[ignore]
-/// Test that signers that have accept a locally signed block N+1 built in tenure A can sign a block proposed during a
+/// Test that signers that have accepedt a locally signed block N+1 built in tenure A can sign a block proposed during a
```
Comment (obycode) [inline testnet/stacks-node/src/tests/signer/v0.rs:4504]: Almost got it 😆

**Review by obycode — APPROVED** [review]

**Pull request merged**